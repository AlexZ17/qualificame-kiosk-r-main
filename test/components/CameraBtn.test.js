import React from 'react';
import { shallow } from 'enzyme';
import { TouchableOpacity } from 'react-native';

import CameraBtn from 'qualificame-kiosk/app/components/CameraBtn';
import { Feather } from '@expo/vector-icons';

function createTestProps (props) {
	return {
		onPress: ()=>{},
		...props,
	}
}

describe('rendering', () => {
	let wrapper;
	const createWrapper = props => shallow(<CameraBtn {...props}/>);
	beforeEach(() => {
		const props = createTestProps()
		wrapper = createWrapper(props);
	});
	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('displays feather icon', () => {
		expect(wrapper.find(Feather).exists()).toBeTruthy();
	});
	it('the feather icon display an camera', () => {
		expect(wrapper.find(Feather).prop('name')).toEqual('camera');
	});
});

describe('interactions', () => {
	let wrapper;
	const createWrapper = props => shallow(<CameraBtn {...props}/>);
	describe('when onPress function is passed to props', () => {
		let fn;
		beforeEach(() => {
			fn = jest.fn();
			const props = createTestProps({ onPress: fn })
			wrapper = createWrapper(props);
		});
		it('it is called on button press', () => {
			wrapper.find(TouchableOpacity).first().props().onPress();
			expect(fn).toBeCalled();
		});
	});
})