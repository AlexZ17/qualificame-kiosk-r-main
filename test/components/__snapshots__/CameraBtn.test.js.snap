// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<TouchableOpacity
  activeOpacity={0.8}
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "#27977F",
      "borderRadius": 50,
      "flexDirection": "row",
      "height": 70,
      "justifyContent": "center",
      "padding": 16,
      "width": 70,
    }
  }
>
  <Icon
    allowFontScaling={false}
    color="white"
    name="camera"
    size={35}
    style={
      Object {
        "alignSelf": "center",
        "width": 35,
      }
    }
  />
</TouchableOpacity>
`;
